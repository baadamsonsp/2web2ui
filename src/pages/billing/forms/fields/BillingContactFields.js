import React from 'react';
import { connect } from 'react-redux';
import { formValueSelector } from 'redux-form';
import PropTypes from 'prop-types';
import CountryField from './CountryField';
import StateProvinceField from './StateProvinceField';
import { Grid } from '@sparkpost/matchbox';
import { TextField } from 'src/components/formFields';
import { required, email } from 'src/helpers/validation';
import { getZipLabel } from 'src/helpers/billing';
import styles from './Fields.module.scss';

export function BillingContactFields(props) {
  return (
    <React.Fragment>
      <pre><code>
        {JSON.stringify(props.currentValues, null, 2)}
      </code></pre>
      <Grid className={styles.spacer}>
        <Grid.Column xs={6}>
          <TextField
            label='First Name'
            name='firstName'
            validate={required}
          />
        </Grid.Column>
        <Grid.Column xs={6}>
          <TextField
            label='Last Name'
            name='lastName'
            validate={required}
          />
        </Grid.Column>
      </Grid>
      <TextField
        label='Email'
        name='email'
        validate={[required, email]}
      />
      <CountryField
        name='country'
        onChange={() => props.change('state', '')}
      />
      <Grid>
        <StateProvinceField
          name='state'
          selectedCountryCode={props.currentValues.country}
          wrapper={({ children }) => <Grid.Column xs={6}>{children}</Grid.Column>}
        />
        <Grid.Column xs={6}>
          <TextField
            label={getZipLabel(props.currentValues.country)}
            name='zip'
            validate={required}
          />
        </Grid.Column>
      </Grid>
    </React.Fragment>
  );
}

BillingContactFields.propTypes = {
  currentValues: PropTypes.object,
  formName: PropTypes.string.isRequired,
  change: PropTypes.func.isRequired
};

// Get country value from state
const mapStateToProps = (state, { formName }) => ({
  currentValues: formValueSelector(formName)(state)
});
export default connect(mapStateToProps)(BillingContactFields);
