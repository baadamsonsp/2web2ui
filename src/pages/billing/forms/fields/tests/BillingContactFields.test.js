import React from 'react';
import { BillingContactFields } from '../BillingContactFields';
import { shallow } from 'enzyme';

describe('Billing Contact Fields:', () => {
  let wrapper;

  const props = {
    formName: 'form-name',
    countryValue: 'GG',
    countries: [
      { value: 'US', label: 'USOFA', states: [{ name: 'mrylnd', value: 'MD' }]}
    ],
    change: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<BillingContactFields {...props} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show states', () => {
    wrapper.setProps({ countryValue: 'US' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle state value', () => {
    const changeSpy = jest.spyOn(wrapper.instance().props, 'change');
    wrapper.instance().handleCountryChange({ target: { value: 'US' }});
    expect(changeSpy).not.toHaveBeenCalled();
    changeSpy.mockReset();
    wrapper.instance().handleCountryChange({ target: { value: 'GG' }});
    expect(changeSpy).toHaveBeenCalledWith(props.formName, 'billingContact.state', null);
  });
});
