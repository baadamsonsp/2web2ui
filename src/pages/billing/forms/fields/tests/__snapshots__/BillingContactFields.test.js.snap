// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billing Address Fields: should render 1`] = `
<div>
  <Grid
    className="spacer"
  >
    <Grid.Column
      xs={6}
    >
      <Field
        component={[Function]}
        label="First Name"
        name="billingContact.firstName"
        validate={[Function]}
      />
    </Grid.Column>
    <Grid.Column
      xs={6}
    >
      <Field
        component={[Function]}
        label="Last Name"
        name="billingContact.lastName"
        validate={[Function]}
      />
    </Grid.Column>
  </Grid>
  <Field
    component={[Function]}
    label="Email"
    name="billingContact.email"
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Country"
    name="billingContact.country"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "USOFA",
          "states": Array [
            Object {
              "name": "mrylnd",
              "value": "MD",
            },
          ],
          "value": "US",
        },
      ]
    }
    placeholder="Select a country"
    validate={[Function]}
  />
  <Grid>
    <Grid.Column
      xs={6}
    >
      <Field
        component={[Function]}
        label="Zip/Postal Code"
        name="billingContact.zip"
        validate={[Function]}
      />
    </Grid.Column>
  </Grid>
</div>
`;

exports[`Billing Address Fields: should show states 1`] = `
<div>
  <Grid
    className="spacer"
  >
    <Grid.Column
      xs={6}
    >
      <Field
        component={[Function]}
        label="First Name"
        name="billingContact.firstName"
        validate={[Function]}
      />
    </Grid.Column>
    <Grid.Column
      xs={6}
    >
      <Field
        component={[Function]}
        label="Last Name"
        name="billingContact.lastName"
        validate={[Function]}
      />
    </Grid.Column>
  </Grid>
  <Field
    component={[Function]}
    label="Email"
    name="billingContact.email"
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Country"
    name="billingContact.country"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "USOFA",
          "states": Array [
            Object {
              "name": "mrylnd",
              "value": "MD",
            },
          ],
          "value": "US",
        },
      ]
    }
    placeholder="Select a country"
    validate={[Function]}
  />
  <Grid>
    <Grid.Column
      xs={6}
    >
      <Field
        component={[Function]}
        label="State"
        name="billingContact.state"
        options={
          Array [
            Object {
              "name": "mrylnd",
              "value": "MD",
            },
          ]
        }
        validate={[Function]}
      />
    </Grid.Column>
    <Grid.Column
      xs={6}
    >
      <Field
        component={[Function]}
        label="Zip Code"
        name="billingContact.zip"
        validate={[Function]}
      />
    </Grid.Column>
  </Grid>
</div>
`;
